src/matrix/lkj.jl

```diff
@@ -188,32 +189,37 @@ end
 
 function lkj_onion_loginvconst(d::Integer, η::Real)
     #  Equation (17) in LKJ (2009 JMA)
-    sumlogs = zero(η)
-    for k in 2:d - 1
-        sumlogs += 0.5k*logπ + loggamma(η + 0.5(d - 1 - k))
+    T = float(Base.promote_typeof(d, η))
+    h = T(1//2)
+    α = η + h * d - 1
+    loginvconst = (2*η + d - 3)*T(logtwo) + (T(logπ) / 4) * (d * (d - 1) - 2) + logbeta(α, α) - (d - 2) * loggamma(η + h * (d - 1))
+    for k in 2:(d - 1)
+        loginvconst += loggamma(η + h * (d - 1 - k))
     end
-    α = η + 0.5d - 1
-    loginvconst = (2η + d - 3)*logtwo + logbeta(α, α) + sumlogs - (d - 2) * loggamma(η + 0.5(d - 1))
     return loginvconst
 end
 
-function lkj_onion_loginvconst_uniform_odd(d::Integer)
+function lkj_onion_loginvconst_uniform_odd(d::Integer, ::Type{T} = Float64) where {T}
     #  Theorem 5 in LKJ (2009 JMA)
-    sumlogs = 0.0
+    sumlogs = zero(T)
     for k in 1:div(d - 1, 2)
-        sumlogs += loggamma(2k)
+        sumlogs += loggamma(T(2k))
     end
-    loginvconst = 0.25(d^2 - 1)*logπ + sumlogs - 0.25(d - 1)^2*logtwo - (d - 1)*loggamma(0.5(d + 1))
+    h = T(1//2)
+    q = T(1//4)
+    loginvconst = q*(d^2 - 1)*logπ + sumlogs - q*(d - 1)^2*logtwo - (d - 1)*loggamma(h*(d + 1))
```

Comment:
Similar to the implementation above we should ensure that be more careful with irrationals and can remove `sumlogs` and the multiplications with `2`:
```suggestion
    h = T(1//2)
    loginvconst = (d - 1) * ((d + 1) * (T(logπ) / 4) - (d - 1) * (T(logtwo) / 4) - loggamma(h * (d + 1)))
    for k in 2:2:(d - 1)
        loginvconst += loggamma(T(k))
    end
```

Reply:
Done. Thanks.